GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m

```diff
@@ -103,19 +103,18 @@ - (void)setObject:(id<NSSecureCoding>)object
 
 - (void)removeObjectForKey:(NSString *)key
                accessGroup:(nullable NSString *)accessGroup
-         completionHandler:(void (^)(BOOL success, NSError *_Nullable error))completionHandler {
+         completionHandler:(void (^)(NSError *_Nullable error))completionHandler {
   dispatch_async(self.inMemoryCacheQueue, ^{
     [self.inMemoryCache removeObjectForKey:key];
     dispatch_async(self.keychainQueue, ^{
       NSDictionary *query = [self keychainQueryWithKey:key accessGroup:accessGroup];
 
       NSError *error;
-      BOOL success = [GULKeychainUtils removeItemWithQuery:query error:&error];
-      if (!success) {
-        completionHandler(NO, error);
-        return;
+      if (![GULKeychainUtils removeItemWithQuery:query error:&error]) {
+        completionHandler(error);
+      } else {
+        completionHandler(nil);
       }
```

Comment:
Compare with original implementation of: 
https://github.com/google/GoogleUtilities/blob/35d94eea3dc1771c8849c1ea5b642fead01370cd/GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m#L117-L123

The implementation of `[GULKeychainUtils removeItemWithQuery:error:]` looks to me like NO is returned iff an error is non-nil.

https://github.com/google/GoogleUtilities/blob/35d94eea3dc1771c8849c1ea5b642fead01370cd/GoogleUtilities/Environment/SecureStorage/GULKeychainUtils.m#L93-L109

Reply:
True, as long as a non-null pointer is passed in, which looks to be the case, despite the Nullable. 

Presumably the ` if (outError) ` tests were redundant in the original code?

Reply:
> Presumably the if (outError) tests were redundant in the original code?

Sorry, I'm not following. That is the current code that the API I'm touching calls into. 

It's a good point about the "True, as long as a non-null pointer is passed in". Since the API I'm touching passes a non-nil address, this should be true when clients use the removeObject API.